enhancements/proposals/1311-replace-nodes-using-hostid/README.md

```diff
@@ -0,0 +1,108 @@
+be empty.
+
+### Procedure flow
+
+#### Existing procedure
+
+Current replace procedure consists of putting a `scylla/replace` label on a node dedicated Service. Operator detects
+that
+the user wants to replace this node, stores the IP address of the node to be replaced in `ScyllaCluster.status.racks[]
+.replace_address_first_boot`.
+After that, Service, Pod and PVC are removed, and new Service is created with a label `scylla/replace` having the IP
+address of the node which is being replaced. Sidecar picks this value and propagates it to the
+`replace-address-first-boot` parameter to ScyllaDB. When Pod becomes Ready, the label is removed from the Service and
+status is cleared.
+
+#### New procedure
+
+Operator will trigger this procedure only when it detects it’s supported by the deployed ScyllaDB version, otherwise it
+will fallback to the old procedure.
+Instead of keeping HostID of a node being replaced in status, will keep the Service intact and sidecar will use HostID
+stored as an annotation to add a replace-node-first-boot parameter to ScyllaDB when the replace label is present.
+ScyllaDB supports replacing a node with the same IP address, this won’t disturb replacing when the Operator is being
+upgraded and versions are skewed.
+
+When a Scylla Operator sees that a replacement label `scylla/replace` is added by the user to the node dedicated Service
+and it’s value is empty, node Pod and associated PVC will be removed, and Service will be updated with
+`internal.scylla-operator.scylladb.com/replacing-node-hostid` with node Host ID as a value.
+When the new Pod comes up, sidecar will propagate the value of HostID annotation from Service into
+`replace-node-first-boot` parameter into ScyllaDB.
+When Service Host ID annotation finally mismatches value stored in the replacement label, and Pod becomes ready, the
+label will be removed ending the procedure.
+
+## Test Plan
+
+E2E tests will verify that both old and new procedures are replacing nodes.
+
+## Upgrade / Downgrade Strategy
+
+Regular Operator upgrade/downgrade strategy.
+
+## Version Skew Strategy
+
+In case when an updated Operator is a leader, it will trigger replacing nodes using host ID.
+When an old Operator would be a leader, it would fill out a `scylla/replace` label with the ClusterIP address of a node
+being replaced and use the old procedure. Sidecar would replace nodes using addresses.
+
+In case when the ScyllaDB version is not high enough, the Operator won’t update the label and won’t delete Pod and PVC.
+In case when the ScyllaDB version would be changed in between Pod is removed and a new one is started, updated sidecar
+would add the node replace parameter anyway. It will cause Scylla to fail, node wouldn’t be replaced, but new one
+wouldn’t join either.
+Users wanting to replace nodes having older ScyllaDB versions will have to upgrade the ScyllaDB version.
```

Comment:
Doesn't that make you stuck potentially? If you have a couple of down nodes and you can't replace them, will you always be able to upgrade the version of your cluster?

Reply:
yep, this will get you stuck on ordered rollouts. I think the answer here is that you'll have to use an older operator until you fix it and can bump a scylla version. then you update the operator

Reply:
This sentence is leftover. Added mention about falling back to old procedure.